lib/ruby/truffle/truffle/bigdecimal.rb

```diff
@@ -181,6 +189,70 @@ def frac
       self
     end
   end
+
+  def to_s(format = 'E')
+    if finite?
+      spaces       = format.to_i
+      prefix       = case
+                     when self > 0
+                       if [' ', '+'].include? format[0]
+                         format[0]
+                       else
+                         ''
+                       end
+                     when self < 0
+                       '-'
+                     when self == 0
+                       ''
+                     end
+      float_format = format[-1] == 'F'
+
+      unscaled_value = unscaled
+      exponent_value = self.exponent
+
+      if float_format
+        before_dot, after_dot = case
+                                when exponent_value > unscaled_value.size
+                                  [unscaled_value + '0' * (exponent_value - unscaled_value.size),
+                                   '0']
+                                when exponent_value <= 0
+                                  ['0',
+                                   '0' * exponent_value.abs + unscaled_value]
+                                else
+                                  after_dot = unscaled_value[exponent_value..-1]
+
+                                  [unscaled_value[0...exponent_value],
+                                   after_dot.empty? ? '0' : after_dot]
+                                end
```

Comment:
splat over a `case`statement with no expression seems too smart :smile:
What about just if/elsif with assigning the two variable in each clause?
```ruby
if exponent_value > unscaled_value.size
  before_dot = unscaled_value + '0' * (exponent_value - unscaled_value.size)
  after_dot = '0'
elsif exponent_value <= 0
  ...
```
(the variables need not be declared before as if/elsif assign in the parent scope)